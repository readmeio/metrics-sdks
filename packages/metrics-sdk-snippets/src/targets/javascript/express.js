import CodeBuilder from '../../helpers/code-builder';

export default function express(params, options) {
  const opts = {
    indent: '  ',
    ...options,
  };

  const code = new CodeBuilder(opts.indent);

  code.push('// Save this code as “server.js”');
  code.push('// Run the server with “node server.js”');
  code.push("const readme = require('readmeio');");
  code.push("const express = require('express');");

  code.blank();

  code.push('const app = express();');
  code.blank();

  code.push("app.post('/webhook', express.raw({ type: 'application/json' }), (req, res) => {");
  code.push(1, '// Verify the request is legitimate and came from ReadMe');
  code.push(1, "const signature = req.headers['readme-signature'];");
  code.push(1, '// Your ReadMe secret');
  code.push(1, "const secret = 'rdme_xxxx';");
  code.push(1, 'try {');
  code.push(2, 'readme.verify(req.body, signature, secret);');
  code.push(1, '} catch (e) {');
  code.push(2, '// Handle invalid requests');
  code.push(2, 'return res.sendStatus(401);');
  code.push(1, '}');

  code.push(1, '// Fetch the user from the db');
  code.push(1, 'db.find({ email: req.body.email }).then(user => {');
  code.push(2, 'return res.json({});');
  code.push(1, '});');
  code.push('});');

  return code.join();
}
