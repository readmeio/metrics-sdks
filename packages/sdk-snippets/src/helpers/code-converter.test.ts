import { describe, it, expect } from 'vitest';

import codeConverter from './code-converter';

const codeSample = `
import express from 'express';

const app = express();

app.get('/', (req, res) => {
  res.sendStatus(200);
});

app.listen(8000);
`;

describe('code-converter', function () {
  it('should convert code to a CodeBuilder instance', function () {
    expect(codeConverter(codeSample)).to.toMatchSnapshot();
  });
});
