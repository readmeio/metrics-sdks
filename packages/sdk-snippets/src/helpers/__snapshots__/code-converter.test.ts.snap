// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`code-converter > should convert code to a CodeBuilder instance 1`] = `
"
blank()

push(\\"import express from 'express';\\");

blank()

push('const app = express();');

blank()

push(\\"app.get('/', (req, res) => {\\");
push('res.sendStatus(200);', 1);
push('});');

blank()

push('app.listen(8000);');

blank()
"
`;
